core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartUpscaleGetStartedInstancesHandler.java

```diff
@@ -0,0 +1,130 @@
+
+    @Inject
+    private CloudInstanceIdToInstanceMetaDataConverter cloudInstanceIdToInstanceMetaDataConverter;
+
+    @Inject
+    private InstanceMetaDataToCloudInstanceConverter instanceMetaDataToCloudInstanceConverter;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartUpscaleGetStartedInstancesRequest> type() {
+        return StopStartUpscaleGetStartedInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartUpscaleGetStartedInstancesRequest> event) {
+        StopStartUpscaleGetStartedInstancesRequest request = event.getData();
+        LOGGER.info("StopStartUpscaleGetStartedInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            Stack stack = stackService.getByIdWithLists(request.getResourceId());
+
+            List<InstanceMetaData> startedInstances = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(
+                    stack,
+                    request.getHostGroupName(),
+                    collectStartedInstancesFromCloudProvider(connector, ac, request.getAllInstancesInHostGroup())
+                            .stream()
+                            .map(CloudInstance::getInstanceId).collect(Collectors.toSet()));
+            ClusterStatusService clusterStatusService = clusterApiConnectors.getConnector(stack).clusterStatusService();
+
+            ExtendedHostStatuses extendedHostStatuses = clusterStatusService.getExtendedHostStatuses(
+                    runtimeVersionService.getRuntimeVersion(stack.getCluster().getId()));
+
+            List<InstanceMetaData> instanceMetadataWithServicesNotRunning = startedInstances.stream()
+                    .filter(i -> i.getInstanceGroupName().equalsIgnoreCase(request.getHostGroupName()))
+                    .filter(i -> !extendedHostStatuses.isHostHealthy(hostName(i.getDiscoveryFQDN())))
+                    .collect(Collectors.toList());
+
+            List<CloudInstance> instancesWithServicesNotRunning = instanceMetaDataToCloudInstanceConverter.convert(instanceMetadataWithServicesNotRunning,
+                    stack.getEnvironmentCrn(), stack.getStackAuthentication());
+
+            StopStartUpscaleGetStartedInstancesResult result =
+                    new StopStartUpscaleGetStartedInstancesResult(request.getResourceId(), request, instancesWithServicesNotRunning,
+                            request.getAdjustment(), request.getHostGroupName());
+            notify(result, event);
+        } catch (Exception e) {
+            String errorMessage = "Failed to collect startedInstances with services not running, from the cloud provider";
```

Comment:
Same as in the case of downscaling - a failure in this step does not mean the entire flow to upscale has failed.